firefox-ios/Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -475,10 +480,17 @@ extension TabScrollingController: UIScrollViewDelegate {
             setOffset(y: 0, for: scrollView)
         }
 
-        let action = GeneralBrowserMiddlewareAction(scrollOffset: scrollView.contentOffset,
-                                                    windowUUID: windowUUID,
-                                                    actionType: GeneralBrowserMiddlewareActionType.didScroll)
-        store.dispatch(action)
+        let scrolledToTop = lastContentOffsetY > 0 && scrollView.contentOffset.y <= 0
+        let scrolledDown = lastContentOffsetY == 0 && scrollView.contentOffset.y > 0
+
+        if scrolledDown || scrolledToTop {
+            lastContentOffsetY = scrollView.contentOffset.y
```

Comment:
Out of curiosity, why do we have to set the `lastContentOffsetY` before and after the dispatch?